heareval/task_embeddings.py

```diff
@@ -72,33 +71,30 @@ def __getitem__(self, idx):
         return self.rows[idx]
 
 
-# These is a high-level wrappers on the basic API that we might
+# High-level wrapper on the basic API that we might
 # consider sharing among all participants.
-
-
 def get_audio_embedding_numpy(
     audio_numpy: np.ndarray,
     model: Any,
     frame_rate: float,
 ) -> Tuple[Dict[int, np.ndarray], np.ndarray]:
-    embedding_dict, timestamps = EMBED.get_audio_embedding(  # type: ignore
+    embedding, timestamps = EMBED.get_audio_embedding(  # type: ignore
         torch.tensor(audio_numpy, device=device),
         model=model,
         frame_rate=frame_rate,
     )
-    for key in embedding_dict.keys():
-        embedding_dict[key] = embedding_dict[key].detach().cpu().numpy()
+    embedding = embedding.detach().cpu().numpy()
     timestamps = timestamps.detach().cpu().numpy()
-    return embedding_dict, timestamps
+    return embedding, timestamps
 
 
 def task_embeddings():
-    model = EMBED.load_model(EMBEDDING_MODEL_PATH, device=device)  # type: ignore
+    model, meta = EMBED.load_model(EMBEDDING_MODEL_PATH, device=device)  # type: ignore
 
     # TODO: Would be good to include the version here
     # https://github.com/neuralaudio/hear2021-eval-kit/issues/37
     embeddir = os.path.join("embeddings", EMBED.__name__)  # type: ignore
-    embedsr = EMBED.input_sample_rate()  # type: ignore
+    embed_sr = meta["sample_rate"]
```

Comment:
Here's another example where doing model.sample_rate seems to make more sense than having a model and a meta

Reply:
On the other hand, I'm wondering if putting it into the class poses some unnecessary restriction on a particular exotic model that I can't imagine right now. I don't think so though